siliconcompiler/schema/schema_obj.py

```diff
@@ -235,16 +229,52 @@ def clear(self, *keypath):
         if not Schema._is_leaf(cfg):
             raise ValueError(f'Invalid keypath {keypath}: clear() must be called on a complete keypath')
 
+        err = Schema._validate_step_index(cfg['pernode'], 'value', step, index)
+        if err:
+            raise ValueError(f'Invalid args to clear() of keypath {keypath}: {err}')
+
         if cfg['lock']:
             # TODO: log here
             return False
 
-        cfg['value'] = cfg['defvalue']
-        cfg['nodevalue'] = {}
-        cfg['set'] = False
+        if step is None and index is None:
+            cfg['nodevalue'] = {}
+            cfg['value'] = cfg['defvalue']
+            cfg['set'] = False
+        else:
+            if index is None:
+                index = 'default'
+            try:
+                del cfg['nodevalue'][step][index]
+            except KeyError:
+                # If this key doesn't exist, silently continue - it was never set
+                pass
 
         return True
 
+    def getvals(self, *keypath):
```

Comment:
I recommend making this a hidden developer function until we work through use cases.